### PR TITLE
NDP References: Clean up Links and AssessmentFile

### PR DESCRIPTION
**Pull request by sorja**
- [x] Move from `assessment/assessmentFile` -> `cycleData/assessmentFile`
- [x] Introduce hooks to get, upload and delete files

Bonus
- [x] Pass generic to `useCountryRouteParams` to avoid casting `countryIso` to `CountryIso` (RFC)

### CODE REVIEW

File: src/client/hooks/useRouteParams.ts [hunk 1]
```diff
@@ -12,7 +13,7 @@ export const useAssessmentRouteParams = () => useParams<AssessmentRouteParams>()
 
 export const useCycleRouteParams = () => useParams<CycleRouteParams>()
 
-export const useCountryRouteParams = () => useParams<CountryRouteParams>()
+export const useCountryRouteParams = <T extends string = AreaCode>() => useParams<CountryRouteParams<T>>()
```
Comment (sorja) [inline src/client/hooks/useRouteParams.ts]: RFC

File: src/meta/routes/routeParams.ts [hunk 2]
```diff
@@ -9,8 +9,8 @@ export type CycleRouteParams = AssessmentRouteParams & {
   cycleName: CycleName
 }
 
-export type CountryRouteParams = CycleRouteParams & {
-  countryIso: AreaCode
+export type CountryRouteParams<T = AreaCode> = CycleRouteParams & {
```
Comment (sorja) [inline src/meta/routes/routeParams.ts:12]: RFC

File: src/client/hooks/useRouteParams.ts (hunk 1 above)
Comment (minotogna) [reply src/client/hooks/useRouteParams.ts]: yeah nice 👍

Comment (minotogna) [reply src/client/hooks/useRouteParams.ts]: actually T can be CountryIso|Region|Global = AreaCode. would that work ?

File: src/client/pages/CountryHome/FraHome/Links/Links.tsx [hunk 3]
```diff
@@ -1,28 +1,28 @@
...
-import { useAssessment, useCycle } from 'client/store/assessment'
-import { AssessmentFilesActions } from 'client/store/ui/assessmentFiles'
-import { useAssessmentFiles } from 'client/store/ui/assessmentFiles/hooks'
+import { useCycle } from 'client/store/assessment'
+import {
+  useAssessmentFiles,
+  useDeleteAssessmentFile,
+  useGetAssessmentFiles,
+  useUploadAssessmentFile,
+} from 'client/store/ui/assessmentFiles/hooks'
```
Comment (minotogna) [inline src/client/pages/CountryHome/FraHome/Links/Links.tsx]: these should be imported from client/store/ui/assessmentFiles

File: src/client/store/ui/assessmentFiles/hooks/index.ts [hunk 4]
```diff
@@ -1,5 +1,68 @@
-import { useAppSelector } from 'client/store'
```
Comment (sorja) [inline src/client/store/ui/assessmentFiles/hooks/index.ts:1]: https://github.com/openforis/fra-platform/pull/3215#discussion_r1390384346

Related to this comment, should we move this to

`src/client/store/data/...`

If so, what would the store look like?
Example?: `state.data[assessmentName][cycleName][countryIso]: Array<AssessmentFile>`
Should we separate the global files to under WO?

File: src/client/store/ui/assessmentFiles/hooks/index.ts [hunk 5]
```diff
@@ -1,5 +1,68 @@
...
+      const uploadProps = { assessmentName, cycleName, countryIso, file, fileCountryIso }
+      const callback = () => {
+        toaster.success(t('landing.links.fileUploaded'))
+      }
+      dispatch(AssessmentFilesActions.upload(uploadProps)).then(callback)
+    },
+    [dispatch, assessmentName, cycleName, countryIso, toaster, t]
+  )
+}
+
```
Comment (minotogna) [inline src/client/store/ui/assessmentFiles/hooks/index.ts]: I would export those from separate files here. and then all of them from assessmentFiles/index

File: src/client/store/ui/assessmentFiles/hooks/index.ts (hunk 4 above)
Comment (minotogna) [reply src/client/store/ui/assessmentFiles/hooks/index.ts:1]: should we do that refactor later on though ?

File: src/client/store/ui/assessmentFiles/hooks/index.ts (hunk 5 above)
Comment (sorja) [reply src/client/store/ui/assessmentFiles/hooks/index.ts]: Sure, let's do so and keep as is otherwise

**Review by minotogna — APPROVED** [review]

**Pull request merged**